### ISSUE TITLE
Proper mask not applied after deleting characters

### ISSUE THREAD

**Issue by MrJox** [RedMadRobot/input-mask-ios#47]
Let have these affineMasks: **[0]**, **[00]**, **[000]**, **[0] [000]**, **[0] [000] [000]**
Let say I have an input:  1 542 (so **[0] [000]** mask applied)
When I remove '2' the input is: 1 54 instead of 154

What can I do to fix this?
I'm using the textField to display currency.
I'm using 3.5.0 version

**Comment by MrJox** [comment]
Yeah it'd have been perfect but for decimal values. Since it's my currency text field, I cannot insert a dot when I need it and when i remove digits from number it automatically puts digits at the end

**Comment by taflanidi** [comment]
@MrJox well, that's unfortunate.
I think our library won't be able to handle your case; at least, current implementation. 

It's a popular user scenario, so I'll be looking into it to be added in the next library versions, though there's a lot of complications to be considered design-wise. Like, I don't want to create yet an another special snowflake mask implementation just for the floating point numbers with digit grouping, I want to make a more broad, flexible & bulletproof solution.
